### PR TITLE
Node attachments

### PR DESCRIPTION
**Pull request by ataulien**
This PR adds the `VisualInteractiveObject` for use with MOBs. It also refactors the `VisualCharacter` so that much code could be put into a generic `VisualSkeletalAnimation` which allows much code to be shared between VisualInteractiveObject` and VisualCharacter`.

It also adds loading of interactive objects from Zen for some simple objects. However, the objects orientations might not be correct. This is because bs:f does not seem to move objects attached to bones to their correct location when no animation is playing...

### CODE REVIEW

**Comment by ataulien** [comment]
Rebased onto master and force-pushed to resolve merge conflicts.

File: src/components/VisualSkeletalAnimation.hpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
+/** \file
```
Comment (mworchel) [inline src/components/VisualSkeletalAnimation.hpp]: Default doxygen comment leftover?

File: src/original-content/VirtualFileSystem.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #include "VirtualFileSystem.hpp"
-#include <Error/BsException.h>
 #include <FileSystem/BsFileSystem.h>
+#include <excepction/Throw.hpp>
```
Comment (mworchel) [inline src/original-content/VirtualFileSystem.cpp]: We should fix that typo soon, before everything get's too big. But that's something for another PR.

**Comment by mworchel** [comment]
Does not compile. You forgot to add and commit main_MobViewer.cpp.

File: src/original-content/VirtualFileSystem.cpp (hunk 2 above)
Comment (ataulien) [reply src/original-content/VirtualFileSystem.cpp]: Oh, that absolutely went past me. We should definitely fix this!

File: src/components/VisualSkeletalAnimation.hpp (hunk 1 above)
Comment (ataulien) [reply src/components/VisualSkeletalAnimation.hpp]: This is needed for the file to be recognized by Doxygen at all. The documentation can be found at the Class then.

**Comment by ataulien** [comment]
That's unfortunate. Usually I just do a `git add .` but this time I had some other files I did not want to commit. You can just remove that sourcefile from the build for now, using the WaynetTester should also work.

**Comment by mworchel** [comment]
Still doesn't compile:
`error C2039: 'ShaderKind': is not a member of 'BsZenLib'`
Did you forget to check in the file where the Enum `BsZenLib::ShaderKind` is defined?

**Comment by ataulien** [comment]
That might be true. I can't update the submodule right now. Can you try to update it manually?

File: src/original-content/VirtualFileSystem.cpp (hunk 2 above)
Comment (mworchel) [reply src/original-content/VirtualFileSystem.cpp]: I can do a PR for that. It's currently only 13 files which have to be fixed (+ the CMakeLists.txt where I forgot to include it in my PR #4).

**Comment by mworchel** [comment]
Compiles and runs after using the master branch of `BsZenLib`. Tested with the `WorldViewer` and `WaynetTester`. Anything specific to look for?

**Comment by ataulien** [comment]
Chests and some other interactive objects should be in the world now.

**Pull request merged**